rosidl_cli/rosidl_cli/extensions.py

```diff
@@ -13,44 +13,86 @@
 # limitations under the License.
 
 import logging
+import re
 
 from rosidl_cli.entry_points import load_entry_points
 
+import yaml
+
 
 logger = logging.getLogger(__name__)
 
 
 class Extension:
     """A generic extension point."""
 
-    def __init__(self, name):
+    def __init__(self, name, **kwargs):
```

Comment:
I think the base class shouldn't have `**kwargs`. I think this in combination with how the PR in rosidl_python is passing remaining args means bogus arguments would be silently ignored. That could be confusing in case of typo's or keyword arguments being deprecated.

```
# The arguments here would be silently ignored
py[typosupport_implementations:[rosidl_typesupport_c]]
```

Reply:
That's a very good point. See https://github.com/ros2/rosidl/pull/597/commits/db920d1e98bcfd899ea0d5a96845f4d03ae275e4.